### PR TITLE
changed merge to update

### LINKED ISSUE #24: The dictionary is never cleared of deleted keys
When a lookup table is loaded from disk, Hash#merge! is used to merge the newly loaded data into the hash used for the lookups. While this overwrites existing keys with the new values, keys that are no longer present in the on-disk file won't be deleted from the in-memory hash and the plugin will happily continue to use the obsolete keys.

### PR DESCRIPTION
**Pull request by xoh**
Entries deleted from a dictionary were not deleted from the internal hash, because `Hash#merge!` was called. Changed this behaviour to really update the dictionary.

Fixes #24

### CODE REVIEW

**Comment by karmi** [comment]
Hi @xoh, we have found your signature in our records, but it seems like you have signed with a different e-mail than the one used in yout Git [commit](https://github.com/logstash-plugins/logstash-filter-translate/pull/57.patch). Can you please add both of these e-mails into your Github profile (they can be hidden), so we can match your e-mails to your Github profile?

**Comment by jsvd** [comment]
@xoh very interesting...we don't document either behaviour. There's no mention that a refresh just updates entries or that it replaces the dictionary as a whole.

I'm tempted to avoid breaking the current behaviour as folks may be relying on it. We could easily provide a setting that defaults to the current behaviour but allows for your need too:

```ruby
config :refresh_behaviour, :validate => ["merge", "replace"], :default => "merge"
```

What do you think?

**Comment by xoh** [comment]
To me it seems odd, that an update can only add, but not remove. But, as you said, some folks may rely on the current functionality and I don't like to break stuff.

I'll add a config setting defaulting to current behaviour and update the PR.

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,3 +1,6 @@
+## 3.0.5
```
Comment (jsvd) [inline CHANGELOG.md]: I suggest bumping this to 3.1.0 as we're adding a new feature.

**Comment by xoh** [comment]
How may I simulate a changing dict-file?

**Comment by xoh** [comment]
Asking differently; I would need to change the internal value of dictionary_path after register:
```
subject.register
subject.filter(first_event)
subject.dictionary_path = altered_dict
subject.filter(second_event)
```
Any idea how to implement this?

**Comment by xoh** [comment]
Old versions of ruby (at least up to the version used by CI with jruby-1.7.27) have poor accuracy of Time.now (just 3 decimal places vs. 9 on current versions). This lead to translate not updating the dictionary_file _sometimes_, but there is/was no actual issue with the filter-code, just with the test-cases. Fixed them, so everything should by ready to merge(?)

**Comment by jsvd** [comment]
@xoh I took the liberty to take your commit and [change the specs a little bit to remove dependency on timing and the extra file](https://github.com/logstash-plugins/logstash-filter-translate/commit/55f70fd09f0e47097436d3d729818115c316e90b).

I merged your commit to master (e33dc0a7c156d85ffde43174210dfaae1d5f19c5) and I'll publish 3.1.0 in a few minutes. Many thanks for your contribution, fast responses and patience.

**Pull request closed without merging**